tools/generate_pv_yaml_rest.sh

```diff
@@ -0,0 +1,328 @@
+                [-c|--storageclass <StorageClass for pv>]
+                [-a|--accessmode <AccessMode for pv>]
+                [-h|--help] " 1>&2
+  exit 1
+}
+
+fullUsage() {
+  echo "Usage: $0
+                -f|--filesystem <Name of Volume's Source Filesystem>
+                -l|--linkpath <full Path of Volume in Primary Filesystem>
+                -F|--fileset <name of source fileset>
+                -s|--size <size in GB>
+                -u|--username <Username of spectrum scale GUI user account.>
+                -t|--password <Password of spectrum scale GUI user account.>
+                -r|--route <Route host name used to route traffic to the spectrum scale GUI service.>
+                [-p|--pvname <name for pv>]
+                [-c|--storageclass <StorageClass for pv>]
+                [-a|--accessmode <AccessMode for pv>]
+                [-h|--help]
+
+
+Example 1: Single Fileystem
+	In this setup there is only one fileystem 'gpfs0' and directory from the same fileystem is being used as volume.
+
+	$0 --filesystem gpfs0 --linkpath /ibm/gpfs0/fileset1/.volumes/staticpv --size 10 --pvname mystaticpv --username admin --password password --route ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.hci-cluster.cp.fyre.ibm.com
+
+
+Example 2: Two or More Filesystem
+	In this setup there are two filesystems 'gpfs0' and 'gpfs1'. gpfs0 is configured as primary fileystem in Spectrum-scale-csi setup. User want to create volume from the directory present in the gpfs1 filesystem. Say the directory in the gpfs1 is /ibm/gpfs1/dir1. As a first step user will create softlink  /ibm/gpfs1/dir1 --> /ibm/gpfs0/fileset1/.volumes/staticpv1 and then run following command to generate the pv.yaml.
+
+	$0 --filesystem gpfs1 --linkpath /ibm/gpfs0/fileset1/.volumes/staticpv1 --size 10 --pvname mystaticpv1 --username admin --password password --route ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.hci-cluster.cp.fyre.ibm.com
+
+Example 3: Fileset based volume
+	This example shows how to create a volume from a fileset 'fileset1' within the filesyetem 'gpfs0'.
+
+	$0 --filesystem gpfs0 --fileset fileset1 --size 10 --pvname mystaticpv --username admin --password password --route ibm-spectrum-scale-gui-ibm-spectrum-scale.apps.hci-cluster.cp.fyre.ibm.com
+
+	Note: This script does not validate if softlinks are correctly created.
+	      The Path specified for option --linkpath must be valid gpfs path from primary fileystem." 1>&2
+  exit 1
+}
+
+# Generate Yaml
+generate_yaml() {
+  volhandle=$1
+  volname=$2
+  volsize=$3
+  accessmode=$4
+  if [[ -f "${volname}.yaml" ]]; then
+    echo "ERROR: File ${volname}.yaml already exist"
+    exit 2
+  fi
+
+  cat >"${volname}".yaml <<EOL
+# -- ${volname}.yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+        name: ${volname}
+spec:
+  capacity:
+    storage: ${volsize}Gi
+  accessModes:
+    - ${accessmode}
+  csi:
+    driver: spectrumscale.csi.ibm.com
+    volumeHandle: ${volhandle}
+  ${STORAGECLASS}
+EOL
```

Comment:
Can you also add PersistentVolumeClaim that makes use of this PV in the yaml?

Reply:
PVC is a namespaced resource, what would be the namespace here? "ibm-spectrum-scale-csi" ?

Reply:
@smitaraut Please verify the sample Generated files in comment below.

Reply:
Should we also change the script name from generate_pv_yaml_rest.sh to something `generate_static_provisioning_yaml.sh` because  generate_pv_yaml_rest.sh gives the perception of only generating pv file ( is my thought but I will upto you.) ?